spec/presenters/tree_builder_alert_profile_obj_spec.rb

```diff
@@ -0,0 +1,126 @@
+describe TreeBuilderAlertProfileObj do
+  before do
+    role = MiqUserRole.find_by(:name => "EvmRole-operator")
+    group = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Tags Group")
+    login_as FactoryGirl.create(:user, :userid => 'tags_wilma', :miq_groups => [group])
+  end
+
+  let!(:tag1a) { FactoryGirl.create(:classification, :name => 'tag1a') }
+  let!(:tag2a) { FactoryGirl.create(:classification, :name => 'tag2a') }
+  let!(:tag3a) { FactoryGirl.create(:classification, :name => 'tag3a') }
+  let!(:folder1a) do
+    f1 = FactoryGirl.create(:classification, :name => 'folder1a', :show => true)
+    f1.entries.push(tag1a)
+    f1.entries.push(tag2a)
+    f1.entries.push(tag3a)
+    f1
+  end
+  let!(:tag1b) { FactoryGirl.create(:tenant, :name => 'tag1b') }
+  let!(:tag2b) { FactoryGirl.create(:tenant, :name => 'tag2b') }
+  let!(:tag3b) { FactoryGirl.create(:tenant, :name => 'tag3b') }
+  let!(:tree_name) { :alert_profile_obj }
+
+  context 'classification tree' do
+    subject do
+      described_class.new(:alert_profile_obj_tree, :alert_profile_obj,
+                          {},
+                          true,
+                          :assign_to => 'storage-tags',
+                          :cat       => folder1a.id,
+                          :objects   => [tag1a.id, tag2a.id])
+    end
+
+    describe '#tree_init_options' do
+      it 'set init options correctly' do
+        expect(subject.send(:tree_init_options, tree_name)).to eq(:expand => true)
+      end
+    end
+
+    describe '#set_locals_for_render' do
+      it 'set locals for render correctly' do
+        locals = subject.send(:set_locals_for_render)
+        expect(locals[:check_url]).to eq("/miq_policy/alert_profile_assign_changed/")
+        expect(locals[:oncheck]).to eq("miqOnCheckHandler")
+        expect(locals[:checkboxes]).to eq(true)
+        expect(locals[:cfmeNoClick]).to eq(true)
+        expect(locals[:onclick]).to eq(false)
+      end
+    end
+
+    describe '#override' do
+      it 'set node' do
+        node = {}
+        subject.send(:override, node, tag1a, nil, nil)
+        expect(node[:hideCheckbox]).to eq(false)
+        expect(node[:select]).to eq(true)
+
+        node = {}
+        subject.send(:override, node, tag2a, nil, nil)
+        expect(node[:select]).to eq(true)
+
+        node = {}
+        subject.send(:override, node, tag3a, nil, nil)
+        expect(node[:select]).to eq(false)
+      end
+    end
+
+    describe '#root_options' do
+      it 'sets root_options correctly' do
+        assign_to = 'storage-tags'
```

Comment:
Is it a good idea? Yes, performance... but this way it is easier to understand the link between the test and the real code. When someone want to change this test or use it as template to test another value of assign_to, he may be confused. What do you think?